petab/simplify.py

```diff
@@ -0,0 +1,106 @@
+"""Functionality for simplifying PEtab problems"""
+from math import nan
+
+import pandas as pd
+
+from . import Problem
+from .C import *  # noqa: F403
+from .lint import lint_problem
+
+__all__ = [
+    "remove_nan_measurements",
+    "remove_unused_observables",
+    "remove_unused_conditions",
+    "simplify_problem",
+    "condition_parameters_to_parameter_table",
+]
+
+
+def remove_nan_measurements(problem: Problem):
+    """Drop any measurements that are NaN"""
+    problem.measurement_df = problem.measurement_df[
+        ~problem.measurement_df[MEASUREMENT].isna()
+    ]
+    problem.measurement_df.reset_index(inplace=True, drop=True)
```

Comment:
Why is this useful? Could be moved to `get_measurement_df` if useful

Reply:
Without that, you will have non-consecutive numbers in the index, which might be somewhat confusing. `get_measurement_df` wouldn't address that.

Reply:
Ah, I meant if it's useful, then should `get_measurement_df` also do it?

Reply:
I don't know if somebody might prefer to keep the original index for whatever reason. For PEtab it wouldn't matter, it's not used anywhere. :man_shrugging: